### PR TITLE
cake 5: Add native type hints for Http properties

### PR DESCRIPTION
**Pull request by othercorey**

### CODE REVIEW

File: src/Http/Client/Response.php [hunk 1]
```diff
@@ -89,35 +89,35 @@ class Response extends Message implements ResponseInterface
...
+    protected ?SimpleXMLElement $_xml = null;
 
     /**
      * Cached decoded JSON data.
      *
-     * @var array
+     * @var mixed
      */
-    protected $_json;
+    protected mixed $_json = null;
```
Comment (othercorey) [inline src/Http/Client/Response.php:120]: Changed to match the function return types.

**Comment by ADmad** [comment]
Just 1 more failing test which sets data as string which causes type error since `ServerRequest::$data` now has type `array|object|null`.

File: tests/TestCase/Http/ServerRequestTest.php [hunk 2]
```diff
@@ -1228,6 +1228,7 @@ public function testGetDataOnStringData(): void
         $request = new ServerRequest(compact('post'));
         $this->assertNull($request->getData('Model'));
         $this->assertNull($request->getData('Model.field'));
+        $this->assertSame($post, $request->getParsedBody());
```
Comment (ADmad) [inline tests/TestCase/Http/ServerRequestTest.php:1231]: Return value of getParsedBody() can't be a string as per the interface https://github.com/php-fig/http-message/blob/efd67d1dc14a7ef4fc4e518e7dee91c271d524e4/src/ServerRequestInterface.php#L164

Comment (ADmad) [reply tests/TestCase/Http/ServerRequestTest.php:1231]: I think we should just throw an exception if the constructor receives as string value as parsed body. We already have the `BodyParserMiddleware` which handles setting the parsed body as array/object from the string body contents https://github.com/cakephp/cakephp/blob/bff920a50e0dd0912e24915a778b0913f57225b2/src/Http/Middleware/BodyParserMiddleware.php#L168.

Comment (markstory) [reply tests/TestCase/Http/ServerRequestTest.php]: Throwing an error if the parsed results are strings makes sense.

Comment (ADmad) [reply tests/TestCase/Http/ServerRequestTest.php]: I have pushed a commit to that effect.

Comment (othercorey) [reply tests/TestCase/Http/ServerRequestTest.php]: This looks good. Can you change the test function description (and name?) just for consistency?

Comment (othercorey) [reply tests/TestCase/Http/ServerRequestTest.php]: n/m, I can do this.

**Pull request merged**